x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/AllocationStatus.java

```diff
@@ -81,7 +81,7 @@ public AllocationStatus(StreamInput in) throws IOException {
     }
 
     public State calculateState() {
-        if (allocationCount == 0) {
+        if (allocationCount == 0 && targetAllocationCount > 0) {
```

Comment:
In case of `targetAllocationCount == 0`, this logic was returning `STARTING`. Change that to `FULLY_ALLOCATED`.

Reply:
Note: this incorrect allocation state also caused ML autoscaling to not scale down, see:
- https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlMemoryAutoscalingDecider.java#L194
- https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingContext.java#L183-L195